### PR TITLE
HDDS-10442. [hsync] Add a Freon tool to measure client to DataNode round-trip latency (#6297)

### PR DESCRIPTION
**Pull request by jojochuang**
(cherry picked from commit 509c970054886c15cd7dd425104be9d0e4e59723)

## What changes were proposed in this pull request?
HDDS-10442. [hsync] Add a Freon tool to measure client to DataNode round-trip latency (#6297)

Please describe your PR in detail:
* Cherrypick HDDS-10442 from feature branch HDDS-7593 to master.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-10442

## How was this patch tested?

Unit tests, Freon integration tests.

### CODE REVIEW

**Comment by jojochuang** [comment]
cc: @kerneltime @duongkame 

Kind of surprised by the amount of conflicts.

**Review by kerneltime — APPROVED** [review]

File: hadoop-hdds/interface-client/src/main/proto/DatanodeClientProtocol.proto [hunk 1]
```diff
@@ -103,6 +103,8 @@ enum Type {
 
   StreamInit = 19;
   StreamWrite = 20;
+  FinalizeBlock = 21;
```
Comment (kerneltime) [inline hadoop-hdds/interface-client/src/main/proto/DatanodeClientProtocol.proto:106]: I think the master is not ready for this field.

**Comment by jojochuang** [comment]
test is green.

**Comment by kerneltime** [comment]
Thanks @jojochuang do this cool extension.

**Pull request merged**